applications/StructuralMechanicsApplication/custom_response_functions/adjoint_elements/adjoint_finite_difference_base_element.h

```diff
@@ -0,0 +1,471 @@
+					      std::vector<double>& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<array_1d<double, 3 > >& rVariable,
+					      std::vector< array_1d<double, 3 > >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<array_1d<double, 6 > >& rVariable,
+					      std::vector< array_1d<double, 6 > >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<Vector >& rVariable,
+					      std::vector< Vector >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<Matrix >& rVariable,
+					      std::vector< Matrix >& rOutput,
+					      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        KRATOS_ERROR << "CalculateOnIntegrationPoints of the adjoint base condition is called!" << std::endl;
+    }
+
+    void GetValueOnIntegrationPoints(const Variable<double>& rVariable,
+					     std::vector<double>& rValues,
+					     const ProcessInfo& rCurrentProcessInfo) override;
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    // Sensitivity functions
+
+    void CalculateSensitivityMatrix(const Variable<double>& rDesignVariable, Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateSensitivityMatrix(const Variable<array_1d<double,3>>& rDesignVariable, Matrix& rOutput,
+                                            const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateStressDisplacementDerivative(const Variable<Vector>& rStressVariable,
+                                    Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo);
+
+    void CalculateStressDesignVariableDerivative(const Variable<double>& rDesignVariable, const Variable<Vector>& rStressVariable,
+                                        Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo);
+
+    void CalculateStressDesignVariableDerivative(const Variable<array_1d<double,3>>& rDesignVariable,
+                                            const Variable<Vector>& rStressVariable,
+                                             Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo);
+
+    Element::Pointer pGetPrimalElement()
```

Comment:
It would be nice if we didn't need this.

Reply:
explained above: needed to have a possibility to revert the replacement process